[prometheus-operator] Rename to kube-prometheus-stack

scottrigby opened the pull request:
Fixes #9 

## Background

See https://github.com/prometheus-community/community/issues/28#issuecomment-670406329

## Description

This is an attempt to rename the chart. There are a _lot_ of changes needed, so it's taken some time to review.

I have also dramatically simplified the README per new charts template decision in #14. (tl;dr, `helm show values` gives all the info a user needs, and makes that the older convention of duplicating these value configurations in the README unnecessary). Without deep experience using this chart, I can't know if the existing upgrading sections are still relevant, so I organized and left them in case they are useful to end users.

## PR status

- [x] All feedback has been addressed so far
- [x] Chart tested both for new installs, and upgrade path from `stable/prometheus-operator`.
    - See https://github.com/scottrigby/prometheus-helm-charts/pull/1#issuecomment-678551180 and https://github.com/scottrigby/prometheus-helm-charts/pull/1#discussion_r474766376 for reasoning for manual testing steps (tl;dr, one of this chart's dependencies has also moved here, so to test installing before we transfer this repo, you need to manually edit `requirements.yaml` and change `https://prometheus-community.github.io/helm-charts` to `https://scottrigby.github.io/prometheus-helm-charts`. The URL will be correct after repo transfer)
- [x] @brancz please address https://github.com/scottrigby/prometheus-helm-charts/pull/1#issuecomment-678617365 by @Xtigyro (`prometheus` chart maintainer) about the new name chosen, and make a final decision.
    - This needs a final say one way or the other. Some alternatives have been suggested, while most others agree `kube-prometheus` is the best name. Be sure to "load more" comments to see the discussion through to https://github.com/scottrigby/prometheus-helm-charts/pull/1#issuecomment-678969511. If you decide to use a different name I will update the PR accordingly
- [x] Rename the chart `kube-prometheus-stack` per https://github.com/prometheus-community/helm-charts/pull/1#issuecomment-687072186

Once steps above are complete, we can merge this PR. Merging will release the new chart package and add to the helm repo index.

**Note:** merging this PR has not been identified as a blocker to transferring the repo to `prometheus-community`. See https://github.com/scottrigby/prometheus-helm-charts/issues/11

vsliouniaev commented:
This seems like it will have issues with PVCs that anyone may have created as part of the previous release as well as anyone using this in umbrella charts. Would be nice to give some guidance on how to solve for those issues using some variables. I've had no time to look at this in the last while.

scottrigby commented:
@vsliouniaev Thanks for looking into this. There is a section in the readme under "Persistent Volumes" that I believe addresses this:
> If you would like to keep the data of the current persistent volumes, it should be possible to attach existing volumes to new PVCs and PVs that are created using the conventions in the new chart… [EXAMPLE]
> The PVC will take ownership of the PV and when you create a release using a persistent volume claim template it will use the existing PVCs as they match the naming convention used by the chart. For other cloud providers similar approaches can be used.

I have not done this myself with this chart, but have helped with migrations by configuring other charts to attach existing claims to the new deployment.

monotek commented:
Wow, I didn't even know that prometheus-operator chart is not the prometheus-operator as I wanted to try out the operator soon.

So +1 to rename it.

scottrigby commented:
Once #14 is merged, will will need to rebase this PR. The change to this chart is deprecating the old chart to begin with so it doesn't get indexed until we finish reviewing this PR and un-deprecate it under the new name.

Xtigyro reviewed (dismissed):
@vsliouniaev @bismarck @scottrigby @zanhsieh 

I find the new name ambiguous. Why renaming this chart?
How will `kube-prometheus` be differentiated from `prometheus`? Do we have `non-kube-prometheus` chart?

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards-1.14/k8s-resources-namespace.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'k8s-resources-namespace' from https://raw.githubusercontent.com/coreos/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'k8s-resources-namespace' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'k8s-resources-namespace' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards-1.14/k8s-resources-cluster.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'k8s-resources-cluster' from https://raw.githubusercontent.com/coreos/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'k8s-resources-cluster' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'k8s-resources-cluster' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards-1.14/controller-manager.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'controller-manager' from https://raw.githubusercontent.com/coreos/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'controller-manager' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'controller-manager' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```

Morriz commented:
For what it's worth, I think `kube-prometheus` matches most closely what it was before, and the idea behind it now. It is the overarching chart bringing the sub parts together for kubernetes users.

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards-1.14/cluster-total.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'cluster-total' from https://raw.githubusercontent.com/coreos/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'cluster-total' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'cluster-total' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards-1.14/apiserver.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'apiserver' from https://raw.githubusercontent.com/coreos/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'apiserver' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'apiserver' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards-1.14/k8s-resources-node.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'k8s-resources-node' from https://raw.githubusercontent.com/coreos/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'k8s-resources-node' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'k8s-resources-node' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards-1.14/k8s-resources-pod.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'k8s-resources-pod' from https://raw.githubusercontent.com/coreos/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'k8s-resources-pod' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'k8s-resources-pod' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards-1.14/k8s-resources-workload.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'k8s-resources-workload' from https://raw.githubusercontent.com/coreos/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'k8s-resources-workload' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'k8s-resources-workload' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards-1.14/k8s-resources-workloads-namespace.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'k8s-resources-workloads-namespace' from https://raw.githubusercontent.com/coreos/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'k8s-resources-workloads-namespace' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'k8s-resources-workloads-namespace' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards-1.14/kubelet.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'kubelet' from https://raw.githubusercontent.com/coreos/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'kubelet' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'kubelet' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards-1.14/namespace-by-pod.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'namespace-by-pod' from https://raw.githubusercontent.com/coreos/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'namespace-by-pod' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'namespace-by-pod' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards-1.14/namespace-by-workload.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'namespace-by-workload' from https://raw.githubusercontent.com/coreos/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'namespace-by-workload' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'namespace-by-workload' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards-1.14/node-cluster-rsrc-use.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'node-cluster-rsrc-use' from https://raw.githubusercontent.com/coreos/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'node-cluster-rsrc-use' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'node-cluster-rsrc-use' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards-1.14/node-rsrc-use.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'node-rsrc-use' from https://raw.githubusercontent.com/coreos/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'node-rsrc-use' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'node-rsrc-use' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards-1.14/nodes.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'nodes' from https://raw.githubusercontent.com/coreos/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'nodes' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'nodes' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards-1.14/persistentvolumesusage.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'persistentvolumesusage' from https://raw.githubusercontent.com/coreos/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'persistentvolumesusage' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'persistentvolumesusage' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards-1.14/pod-total.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'pod-total' from https://raw.githubusercontent.com/coreos/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'pod-total' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'pod-total' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards-1.14/prometheus-remote-write.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'prometheus-remote-write' from https://raw.githubusercontent.com/coreos/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'prometheus-remote-write' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'prometheus-remote-write' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards-1.14/prometheus.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'prometheus' from https://raw.githubusercontent.com/coreos/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'prometheus' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'prometheus' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards-1.14/proxy.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'proxy' from https://raw.githubusercontent.com/coreos/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'proxy' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'proxy' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards-1.14/scheduler.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'scheduler' from https://raw.githubusercontent.com/coreos/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'scheduler' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'scheduler' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards-1.14/statefulset.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'statefulset' from https://raw.githubusercontent.com/coreos/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'statefulset' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'statefulset' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards-1.14/workload-total.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'workload-total' from https://raw.githubusercontent.com/coreos/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'workload-total' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'workload-total' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/master/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards/statefulset.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'statefulset' from https://raw.githubusercontent.com/coreos/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'statefulset' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'statefulset' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards/pods.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'pods' from https://raw.githubusercontent.com/coreos/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'pods' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'pods' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards/persistentvolumesusage.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'persistentvolumesusage' from https://raw.githubusercontent.com/coreos/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'persistentvolumesusage' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'persistentvolumesusage' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards/nodes.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'nodes' from https://raw.githubusercontent.com/coreos/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'nodes' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'nodes' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards/k8s-resources-workloads-namespace.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'k8s-resources-workloads-namespace' from https://raw.githubusercontent.com/coreos/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'k8s-resources-workloads-namespace' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'k8s-resources-workloads-namespace' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards/k8s-resources-workload.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'k8s-resources-workload' from https://raw.githubusercontent.com/coreos/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'k8s-resources-workload' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'k8s-resources-workload' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards/k8s-resources-pod.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'k8s-resources-pod' from https://raw.githubusercontent.com/coreos/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'k8s-resources-pod' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'k8s-resources-pod' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards/k8s-resources-namespace.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'k8s-resources-namespace' from https://raw.githubusercontent.com/coreos/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'k8s-resources-namespace' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'k8s-resources-namespace' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards/k8s-resources-cluster.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'k8s-resources-cluster' from https://raw.githubusercontent.com/coreos/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'k8s-resources-cluster' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'k8s-resources-cluster' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards/k8s-node-rsrc-use.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'k8s-node-rsrc-use' from https://raw.githubusercontent.com/coreos/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'k8s-node-rsrc-use' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'k8s-node-rsrc-use' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
```

omegas27 commented on charts/kube-prometheus/templates/grafana/dashboards/k8s-cluster-rsrc-use.yaml:
```diff
@@ -1,23 +1,23 @@
 {{- /*
-Generated from 'k8s-cluster-rsrc-use' from https://raw.githubusercontent.com/coreos/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
+Generated from 'k8s-cluster-rsrc-use' from https://raw.githubusercontent.com/kube-prometheus/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
```
```suggestion
Generated from 'k8s-cluster-rsrc-use' from https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/release-0.1/manifests/grafana-dashboardDefinitions.yaml
```

scottrigby replied:
Hi @omegas27 thanks for fixing that. I made all 83 changes in one commit: 6356d02 and listed you as a co-author 👍 

**For everyone else following:** this is a non-functional typo fix to commented out CRD URLs 🙂📝

omegas27 replied:
This is definitely better than 83 commits :D

scottrigby replied:
Repo settings are for PRs to be squashed anyway, but yeah 😅

torstenwalter commented:
@scottrigby I messed it up when merging the PR you had approved. I improved the commit message and forgot the sign off that commit in. Therefor I had to force push the last commit to fix DCO.

brancz commented:
@Xtigyro how about prometheus-operator-kube-prometheus then? That would remove the ambiguity. (to be fair I have the same problem with the prometheus chart being a lot more than just Prometheus, as the "current" prometheus-operator chart)

Xtigyro commented:
@brancz That name is very clear. However, isn't it too long? I have no problem with that. I just want a fair choice for you as well so you are happy too. That's why I proposed `kube-prometheus-distro`.

mrueg commented:
I guess since it's a chart, we might be able to "omit" kube- completely. How about `prometheus-stack`?

brancz commented:
I want to make it crystal clear that it's based on the upstream kube-prometheus project, I understand that helm sets the kubernetes context, but it's more about association with the "upstream" project.

scottrigby commented:
Personally I admit I had never understood that this chart installs the [kube-prometheus](https://github.com/prometheus-operator/kube-prometheus) project. It appeared to be a generally opinionated wrapper for the [prometheus-operator](https://github.com/prometheus-operator/prometheus-operator) project.

And I'm not only a user who had installed `stable/prometheus-operator` many times, I'm also a Helm maintainer and maintain the `stable` and `incubator` charts repos. I have even met with @gianrubio and discussed/installed the `stable/prometheus` chart in person. So I'm guessing that I was confused, a large percent of the user base is also likely unclear about this. If I understand correctly, this is why @brancz made this naming the primary condition for sponsoring bringing the prometheus charts into the `prometheus-community` GitHub org.

scottrigby commented:
Because contributors PRs in `stable` are on hold until this is complete, I would suggest agreeing on the name ASAP.

**So where do we stand:**
- There was one dissenting view about `kube-prometheus`, by @Xtigyro (concerned end users may mistake `kube-prometheus` for the `prometheus` chart he maintains. Fair enough)
- @brancz has made clear that `kube-prometheus` must be clearly part of the name (because of the long-standing issue above. Example `prometheus-operator-kube-prometheus` or something else containing the project name. Very fair)
- Other names suggested: `kube-prometheus-distro` (@Xtigyro) and `prometheus-stack` (@mrueg)

**Suggestion:**
How about we combine the above suggestions to be **`kube-prometheus-stack`**?

I just checked and the upstream [kube-prometheus](https://github.com/prometheus-operator/kube-prometheus) project. While there is no reference to "distro", theh [README](https://github.com/prometheus-operator/kube-prometheus) has 5 references to "kube-prometheus stack" (such as "The kube-prometheus stack includes…"), and 13 references to the project "stack" overall (such as "This stack provides…").

@brancz would this be acceptable?

Morriz commented:
Then just how about `prometheus-stack`, as it is a prometheus related bundle? We all know it's for k8s by now, no need to carry that in the name, right?

brancz commented:
I am happy with `kube-prometheus-stack`.

monotek commented:
"kube-prometheus-stack" sound good to me too.

Xtigyro commented:
Very nice - thanks, @brancz!

scottrigby commented:
The name change is pushed. I'm testing this locally myself, and everything seems to be running smoothly. 

**How you can help**
Can others please test this and review? This is the last thing I believe we need before all the stable prometheus charts will have been successfully migrated 😄

scottrigby commented:
CI passed! 👍

zanhsieh commented:
I am waiting for this merge. There's a PR Lint and Test Charts / lint-test (pull_request) failure due to this:

https://github.com/prometheus-community/helm-charts/pull/49

scottrigby commented:
I'll resolve the new merge conflict tomorrow.

Also @bismarck said he will test the adopting existing PVCs use case this week. Anyone else is welcome to test that as well, if you'd like to help.

If that tests ok, we will need an approving review by one of the two maintainers in order to merge.

Xtigyro commented:
Two small fixes here: https://github.com/prometheus-community/helm-charts/pull/60

gkarthiks commented:
@scottrigby @torstenwalter @gianrubio I don't understand what is this failing for in DCO. Can any one of you explain this pls?

gianrubio commented on charts/kube-prometheus-stack/Chart.yaml:
```diff
@@ -0,0 +1,22 @@
+  - name: gianrubio
+    email: gianrubio@gmail.com
+  - name: gkarthiks
+    email: github.gkarthiks@gmail.com
```
Would you mind adding yourself as a maintainer ?

gkarthiks replied:
@gianrubio I guess you are asking @scottrigby right?

gianrubio replied:
> @gianrubio I guess you are asking @scottrigby right?

Exactly!

scottrigby replied:
Added

gianrubio commented:
I'm sorry! I've been working on other projects in the past year, therefore I could not be involved with supporting the helm charts. 
I fully agree with the chart renaming. 

One thing that is missing here is a note explaining that this chart can break the compatibility from helm/charts. It's fine to not have a clear migration path from the previous repo as long all users are informed. Hopefully, the community will find an easy way to migrate from there and it will send a PR :)

Finally, thanks for the community for supporting this helm chart!

scottrigby commented:
Ignore the fun new conflicts - will fix those in a bit. Rebased & force-with-lease in order to fix DCO. I'll link the issue later

scottrigby commented:
Fixed DCO. @gkarthiks The reason was the author/signoff in e729fdb.  See https://github.com/probot/dco/issues/134. I amended your commit with signoff formatted the same as your author commit info. I believe in this case I have your permission do do that, but correct me if I'm wrong 🙂

I had to force-push to fix DCO, so some comment connections may be lost in the UI. But @gianrubio I added myself as you suggested. Everything else should be the same as the accepting reviews. The one thing I want o

scottrigby commented:
Ok 👌 tests are passing. DCO fixed. After changes pushed, needs another accepting review from someone in the prometheus-operator maintainers list again - and then we can finally merge this 😅

desaintmartin commented on charts/kube-prometheus-stack/README.md:
```diff
@@ -0,0 +1,312 @@
+
+_See [helm upgrade](https://helm.sh/docs/helm/helm_upgrade/) for command documentation._
+
+### Migrating from stable/prometheus-operator chart
```
I think some lines are missing here.

gkarthiks replied:
@desaintmartin I see all the 312 lines from original file.

desaintmartin replied:
Oh, I understand. It may be clearer to keep
```suggestion
### Upgrading an existing Release to a new major version

Note: versions < 9.3.3 were published to the `stable/prometheus-operator` chart
```

gkarthiks replied:
aah, makes sense.

scottrigby replied:
done

torstenwalter commented:
Nice! We are getting there. Two approvals. @scottrigby you should be the one to do the merge ;-)

steven-sheehy commented:
I'm not sure why the need to prefix with kube as all Helm charts are by definition associated with kubernetes already. So within that context prefixing with kube is redundant. I prefer the simpler `prometheus-stack` as it:
- Drops the unnecessary prefix
- Matches the prefix of every other chart in this repo (e.g. `prometheus-`)
- Clearly differentiates between the separate [prometheus-operator](https://github.com/prometheus-operator/prometheus-operator) _and_ [kube-prometheus](https://github.com/prometheus-operator/kube-prometheus) projects
- Reads nicer: Indicates you're installing prometheus plus a stack of other components. You're not installing a "kube".

scottrigby commented:
@steven-sheehy I know a ton of discussion is buried in GitHub collapsed comments, so please read the direct links: 
- Summary: https://github.com/prometheus-community/helm-charts/pull/1#issuecomment-686495216
- Final sponsor approval: https://github.com/prometheus-community/helm-charts/pull/1#issuecomment-687072186

The `kube-prometheus` project name _must_ be part of the chart name, as an absolute condition of moving this git repo to the `prometheus-community` org. The maintainers of that chart and prometheus devs (including maintainer of upstream `kube-prometheus` project) have agreed on the name. That process is now finished.

scottrigby commented:
I believe we've covered all our bases and then some over the past month of this PR. We're now – finally – ready to merge 😅  With that said… Thanks everyone! 🎉 💖 🙌 

![press the button](https://i.giphy.com/media/ZEr5tiKWdvpfYa2L9V/giphy.webp)

Xtigyro commented:
@scottrigby Thank you, man! Your work has been UNBELIEVABLY VALUABLE! 🥇
